controllers/storagecluster/reconcile.go

```diff
@@ -212,6 +212,12 @@ func (r *StorageClusterReconciler) validateStorageClusterSpec(instance *ocsv1.St
 		r.Log.Error(err, "Failed to validate StorageCluster version.", "StorageCluster", klog.KRef(instance.Namespace, instance.Name))
 		r.recorder.ReportIfNotPresent(instance, corev1.EventTypeWarning, statusutil.EventReasonValidationFailed, err.Error())
 		instance.Status.Phase = statusutil.PhaseError
+		statusutil.SetStatusConditionIfNotPresent(&instance.Status.Conditions, conditionsv1.Condition{
+			Type:    conditionsv1.ConditionDegraded,
+			Status:  corev1.ConditionTrue,
+			Reason:  statusutil.EventReasonValidationFailed,
+			Message: err.Error(),
+		})
```

Comment:
Not sure if Degraded is the right status here. We should have a condition for version mismatch.

Also, no need to export `SetStatusConditionIfNotPresent`. Just define a new public function in statusutil to set version mismatch status.

Reply:
Ack, will do that.